src/compiler/scala/tools/nsc/Global.scala

```diff
@@ -159,16 +159,18 @@ class Global(var currentSettings: Settings, var reporter: Reporter)
   // Components for collecting and generating output
 
   import scala.reflect.internal.util.StatisticsInfo
-  trait AllStats extends ReflectStats
-                    with TypersStats
-                    with ImplicitsStats
-                    with MacrosStats
-                    with BackendStats
+  import scala.tools.nsc.transform.patmat.PatternMatchingStats
+  trait GlobalStats extends ReflectStats
+                       with TypersStats
+                       with ImplicitsStats
+                       with MacrosStats
+                       with BackendStats
+                       with PatternMatchingStats
```

Comment:
How do you feel about removing all those traits and just moving the stats into the `GlobalStats` / `ReflectStats`? I don't see much value in having these definitions spread around. Now that all stats are mixed into a single class, the definition site no longer acts as a namespace.

Reply:
I kind of like the way it is right now not because they have their own namespace but because they are located right next to the place they're used -- so if I'm interested in macro statistics I can just grep through the file and get to their definition.
  
But if you think this is cumbersome, I have no problem to move all of them to those concrete trait. I'm really not a fan of the cake.

Reply:
I think it would make the whole thing easier to read, but I don't have very strong feelings about it either.

Reply:
I propose to keep it the way it is and if other person complains I'll put it together. Personally, I don't find it hard to read, I just dislike the self-type dependencies... In any case, my last commit removed `StatisticsInfo`, so the last version is cleaner.